syntax/compile.go

```diff
@@ -495,7 +495,13 @@ func (pc ParseContext) compileExpr(c ast.Children) rel.Expr {
 		if len(c) == 1 {
 			return pc.CompileExpr(c[0].(ast.Branch))
 		}
-		panic("too many expr children")
```

Comment:
I am not sure if this change will break the initial design, can you help to double confirm?

Reply:
You may be able to avoid this hack by moving the logic into `compileCond`.